frontend/src/components/SelectedEventModal.tsx

```diff
@@ -15,39 +16,55 @@ import {
     HStack,
     VStack,
     Text,
-    // ListItem,
-    // UnorderedList,
 } from '@chakra-ui/react';
+import moment from 'moment';
 import { getEventMonthDaySpan, getEventStartAndEndTime } from '../controller/utils';
-// import { deleteJob } from '../controller/job';
-// import { deleteEvent } from '../controller/event';
+import { UserContext } from '../contexts/UserContext';
+import { Job, PatchJobRequest } from '../../../src/typedefs/job';
+import { Event } from '../../../src/typedefs/event';
 
 interface modalProps {
   isOpen: boolean,
   onClose: () => void,
   selectedEvent: any,
   onSignUpOpen: () => void;
-  // attendeesList: any[], // TODO this should match our typing
-  admin: boolean
+  admin: boolean;
+  deleteEvent: () => void;
+  // eslint-disable-next-line no-unused-vars
+  signUpForJob: (patchInfo: { jobId: number; editedJob: PatchJobRequest; }) => void
 }
 
-async function handleSignUp(selectedJob: any) {
-    // eslint-disable-next-line no-console
-    console.log(selectedJob);
-}
+export default function SelectedEventModal(props: modalProps) {
+    const { state } = useContext(UserContext);
 
-async function deleteEventLocal(event: any) {
-    // if (event.type === 'work') {
-    //     await deleteJob('TestingToken', event);
-    // } else {
-    //     await deleteEvent('TestingToken', event);
-    // }
-    // eslint-disable-next-line no-console
-    console.log(event);
-    // Update state?
-}
+    function isJob(selectedEvent: Event | Job): selectedEvent is Job {
+        if ((selectedEvent as Job).jobId) {
+            return true;
+        }
+        // else, its an Event
+        return false;
+    }
+
+    async function generateJobSignUpPatch() {

```

Comment:
this looks to be duplicated logic as well

Reply:
What are your thoughts on putting this in the job controller? Utils?